contracts/bond/CollateralWhitelist.sol

```diff
@@ -3,15 +3,25 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableMapUpgradeable.sol";
 
 /**
  * @title Whitelist for collateral tokens.
  *
  * @notice Encapsulation of a ERC20 collateral tokens whitelist, indexed by their symbol.
  */
 abstract contract CollateralWhitelist is Initializable {
+    using EnumerableMapUpgradeable for EnumerableMapUpgradeable.UintToAddressMap;
+
     // Token symbols to ERC20 Token contract addresses
-    mapping(string => address) private _whitelist;
+    EnumerableMapUpgradeable.UintToAddressMap private _whitelist;
+
+    /**
+     * @notice Returns an array of all currently whitelisted symbols
+     */
+    function whitelistedSymbols() external view returns (bytes32[] memory) {
```

Comment:
this should be changed to return string[] memory

Reply:
Depending on what exactly the `bytes32` is storing, the [OZ String utiles](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/429be5f120309f4faf8b2bea4c242e1b87d22f9a/contracts/utils/StringsUpgradeable.sol#L15) may help?

Reply:
The bytes32 here is storing the key for the mapping - in our case the symbol string